mod/da/pkg/blob/verifier.go

```diff
@@ -72,18 +76,26 @@ func (bv *Verifier[_, _, BlobSidecarsT]) VerifySidecars(
 		bv.proofVerifier.GetImplementation(),
 	)
 
+	// check that sideracs block headers match with header of the
+	// corresponding block
+	for i, s := range sidecars.GetSidecars() {
+		if !reflect.DeepEqual(s.GetBeaconBlockHeader(), blkHeader) {
```

Comment:
are we sure `reflect.DeepEqual` is good here? otherwise might make sense to add a Equality function on the blockHeader type

Reply:
replaced with equals (and added simple UTs to avoid surprises)